### PR TITLE
1763 username gen

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/1994
Replaces https://github.com/backdrop/backdrop/pull/1763

### CODE REVIEW

File: core/modules/user/user.module [hunk 1]
```diff
@@ -3086,6 +2986,25 @@ function user_register_submit($form, &$form_state) {
 }
 
 /**
+ * Implements hook_user_insert().
+ */
+function user_user_insert($account) {
+  // Don't create a new username if one is already set.
+  if (!empty($account->name) && strpos($account->name, 'account-') !== 0) {
```
Comment (docwilmot) [inline core/modules/user/user.module:2993]: What happens if usernames are allowed and someone chooses 'account-manager'?

File: core/modules/user/user.module [hunk 2]
```diff
@@ -879,49 +878,39 @@ function user_account_form_validate($form, &$form_state) {
...
+        if (config('system.core')->get('user_email_match') && (valid_email_address($name))) {
+          if ($name !== $mail) {
+            form_set_error('name', t('An email address was provided as a username, but does not match the account email address.'));
+          }
         }
       }
     }
+    else {
+      // Here we can use user_password() to generate a safe random string.
+      // This username will be updated t use the UID in hook_user_insert().
```
Comment (docwilmot) [inline core/modules/user/user.module:911]: *to*

File: core/modules/user/user.module [hunk 3]
```diff
@@ -879,49 +878,39 @@ function user_account_form_validate($form, &$form_state) {
...
+      $form_state['values']['name'] = 'account-' . user_password();
```
Comment (docwilmot) [inline core/modules/user/user.module:912]: if this is temporary, use a more unguessable string than 'account-'. its not unlikely that a real user would choose that as the prefix of his username. 'tmpacctstrng'?

File: core/modules/user/user.module (hunk 1 above)
Comment (jenlampton) [reply core/modules/user/user.module:2993]: Good point! Maybe we should put something longer and more likely to be unique in as a placeholder...

File: core/modules/user/user.module (hunk 3 above)
Comment (jenlampton) [reply core/modules/user/user.module:912]: yes, good call.

**Comment by jenlampton** [comment]
closing in favor of an alternate solution. See https://github.com/backdrop/backdrop-issues/issues/2669

**Pull request closed without merging**